### PR TITLE
Sdf log

### PR DESCRIPTION
**Pull request by rauldds**

### CODE REVIEW

File: inference/object_completion_new.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    voxels = np.clip(voxels, -3, 3).reshape((1, 1, 64, 64, 64))
+    print(voxels.shape)
+    voxels = torch.from_numpy(voxels).cuda()
+    model.eval()
+
+
+    # Perform the inference
+    with torch.no_grad():
+        recon, skip = model.forward(data=voxels)
+        recon = torch.exp(recon) - 1.0
```
Comment (rauldds) [inline inference/object_completion_new.py:60]: Same as previous comment
```suggestion
        recon =recon
```

File: inference/object_completion.py [hunk 2]
```diff
@@ -37,6 +37,8 @@
 
 recon, skip = model.forward(data=voxels)
 
+recon = torch.exp(recon) - 1.0
```
Comment (rauldds) [inline inference/object_completion.py:40]:
```suggestion
recon = recon
```
Why are we doing this? The output of the network is supposed. This would make sense to me if we directly inside the network applied the log and +1. However we do that outside the network in the training loop. So this doesn't make too much sense to me

File: model/train_grnet.py [hunk 3]
```diff
@@ -212,16 +227,16 @@ def train(model_comp, model_clas, train_dataloader, val_dataloader,
...
 
 
-                tb.add_scalar("Val_Loss", val_loss, epoch)
+                tb.add_scalar("Val_Loss", val_loss/(batch_val["incomplete_view"].shape[0] * len(val_dataloader)), epoch)
 
 
 
         #Path(f'/ckpts').mkdir(exist_ok=True, parents=True)
-        batch_loss = batch_loss/len(train_dataloader)
+        val_loss /= len(val_dataloader)
```
Comment (rauldds) [inline model/train_grnet.py]: @DL-Eyes you said you divided by the number of samples right? this is dividing by the number of batches. That's why when I created batch_loss I gave it that name

File: model/train_grnet.py [hunk 4]
```diff
@@ -12,10 +12,24 @@
...
+    def __init__(self, weight=None, size_average=True):
+        super(log_space_L1_loss, self).__init__()
+
+    def forward(self, reconstruction, target):
+        target_log = torch.log(torch.abs(target) + 1)
+
+        loss = torch.abs(torch.abs(reconstruction) - target_log)
+
+        sign_mask = torch.sign(reconstruction) * torch.sign(target) < 0
+        loss = torch.where(sign_mask, 2 * loss, loss)
```
Comment (rauldds) [inline model/train_grnet.py:26]: Why are we making 2*loss for the places where both locations (target and reconstruction) are inside and just loss for the rest (when they have different signs)?

Comment (Lguoad) [reply model/train_grnet.py:26]: Yes, you are correct, the mask trick will make it meaningless, so maybe only keep line23?

File: inference/object_completion.py (hunk 2 above)
Comment (Lguoad) [reply inference/object_completion.py:40]: I just follow what E3 they do to 3depn network. I think the output of 3depn network is in log space
